### PR TITLE
passport authentication, profile and recipe routes draft - Val

### PR DESCRIPTION
**Pull request by ValEmpire**
Passport auth, profile and recipe routes - all routes, middlewares, and controllers are not tested, just want to show data structure, flow and algorithm

### CODE REVIEW

File: server/controllers/profile.controllers.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        cuisineTags,
+      } = req.body;
+
+      // if user is not the owner of the profile
+      if (user._id !== profile.user._id)
+        throw "Cannot update another user profile";
+
+      // update profile
+      // Wiill change the location field
+      // Still figuring out how to save coordinates
```
Comment (rudream) [inline server/controllers/profile.controllers.js:85]: Looks good! Don't worry about dealing with the coordinates, I'll be implementing the geocoding logic

Comment (rudream) [inline server/controllers/profile.controllers.js:85]: Don't worry about dealing with the coordinates, I'll be implementing the geocoding logic 😃

File: server/controllers/profile.controllers.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
+const Profile = require("../models/profile.model");
+
+module.exports = {
```
Comment (rudream) [inline server/controllers/profile.controllers.js:3]: I would recommend that you define the functions outside of this and then export them at the end

eg. 

```
const createProfile = async (req, res) => {...}

const retrieveProfile: async (req, res) => {...}

exports.createProfile = createProfile;
exports.retrieveProfile = retrieveProfile

```
This looks better in my opinion, and is consistent with the way @Vaibhav-Kambli has already set it up (so it won't cause merge conflicts)

File: server/controllers/profile.controllers.js (hunk 1 above)
Comment (ValEmpire) [reply server/controllers/profile.controllers.js:85]: That's great!. Thank you.

File: server/controllers/profile.controllers.js (hunk 2 above)
Comment (ValEmpire) [reply server/controllers/profile.controllers.js:3]: I will refactor it. Thank you. 👍

**Pull request closed without merging**